### PR TITLE
Create Docker image

### PR DESCRIPTION
**Pull request by apulsipher**
- Create Docker image to use for running the forecast
- Create GitHub Action to automatically build and publish the image to GH Container Registry (set to run manually or whenever the Dockerfile changes)
- Update README with documentation on changes
- Create `.devcontainer/` for use in VSCode

### CODE REVIEW

File: .devcontainer/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Use rocker/tidyverse as a base to get: 
+# - R, quarto, data.table, ggplot2
+FROM rocker/tidyverse:4.4.1
+
+# Install epiworldR
+RUN install2.r epiworldR
```
Comment (gvegayon) [inline .devcontainer/Dockerfile:6]: I usually end my Dockerfiles with `CMD ["bash"]` don't we need that here? Maybe not?!

File: .github/workflows/build-docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      packages: write
+      attestations: write
+      id-token: write
+      # 
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      # Uses the `docker/login-action` action to log in to the Container registry registry using the account and password that will publish the packages. Once published, the packages are scoped to the account defined here.
+      - name: Log in to the Container registry
+        uses: docker/login-action@65b78e6e13532edd9afa3aa52ac7964289d1a9c1
```
Comment (gvegayon) [inline .github/workflows/build-docker-image.yml:34]: Do we want to use a targeted commit?

File: .github/workflows/build-docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      # This step uses [docker/metadata-action](https://github.com/docker/metadata-action#about) to extract tags and labels that will be applied to the specified image. The `id` "meta" allows the output of this step to be referenced in a subsequent step. The `images` value provides the base name for the tags and labels.
+      - name: Extract metadata (tags, labels) for Docker
+        id: meta
+        uses: docker/metadata-action@9ec57ed1fcdbf14dcef7dfbe97b2010124a938b7
```
Comment (gvegayon) [inline .github/workflows/build-docker-image.yml:42]: Same

File: .github/workflows/build-docker-image.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+        with:
+          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+      # This step uses the `docker/build-push-action` action to build the image, based on your repository's `Dockerfile`. If the build succeeds, it pushes the image to GitHub Packages.
+      # It uses the `context` parameter to define the build's context as the set of files located in the specified path. For more information, see "[Usage](https://github.com/docker/build-push-action#usage)" in the README of the `docker/build-push-action` repository.
+      # It uses the `tags` and `labels` parameters to tag and label the image with the output from the "meta" step.
+      - name: Build and push Docker image
+        id: push
+        uses: docker/build-push-action@f2a1d5e99d037542a71f64918e516c093c6f3fc4
```
Comment (gvegayon) [inline .github/workflows/build-docker-image.yml:50]: Same

File: .github/workflows/build-docker-image.yml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+          tags: ${{ steps.meta.outputs.tags }}
+          labels: ${{ steps.meta.outputs.labels }}
+      
+      # This step generates an artifact attestation for the image, which is an unforgeable statement about where and how it was built. It increases supply chain security for people who consume the image. For more information, see "[AUTOTITLE](/actions/security-guides/using-artifact-attestations-to-establish-provenance-for-builds)." 
+      - name: Generate artifact attestation
+        uses: actions/attest-build-provenance@v1
+        with:
+          subject-name: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}
+          subject-digest: ${{ steps.push.outputs.digest }}
+          push-to-registry: true
```
Comment (gvegayon) [inline .github/workflows/build-docker-image.yml:63]: This is the first time I have seen this. Does this information go somewhere?

File: README.md [hunk 6]
```diff
@@ -12,4 +12,11 @@ The forecast will update automatically each week using GitHub Actions.
...
-In the future, we may use Germ Watch or additional public data sources.
\ No newline at end of file
+In the future, we may use Germ Watch or additional public data sources.
+
+## Building Container Image
+Use the Dockerfile in [`.devcontainer/`] to build the `epiworld-forecasts` image. 
+The image includes [`rocker/tidyverse`](https://rocker-project.org/images/versioned/rstudio.html) and [`epiworldR`](https://github.com/UofUEpiBio/epiworldR). 
+It is stored on the Github Container Registry (`ghcr.io/`).
+
+**Note for macOS users:** Pulling and building the `rocker/tidyverse` image require specifying the platform with `--platform=linux/amd64`
```
Comment (gvegayon) [inline README.md:22]:
```suggestion
**Note for macOS users (M chips):** Pulling and building the `rocker/tidyverse` image may require specifying the platform with `--platform=linux/amd64`
```

**Review by gvegayon — COMMENTED** [review]
Looks good! A couple of questions before merging.

File: .devcontainer/Dockerfile (hunk 1 above)
Comment (apulsipher) [reply .devcontainer/Dockerfile:6]: Good catch. I added it.

File: README.md (hunk 6 above)
Comment (apulsipher) [reply README.md:22]: Added.

File: .github/workflows/build-docker-image.yml (hunk 2 above)
Comment (apulsipher) [reply .github/workflows/build-docker-image.yml:34]: I looked into this, and GitHub recommends it as a good security practice, described [here](https://docs.github.com/en/actions/security-for-github-actions/security-guides/security-hardening-for-github-actions#using-third-party-actions):

> ### Pin actions to a full length commit SHA
> Pinning an action to a full length commit SHA is currently the only way to use an action as an immutable release. Pinning to a particular SHA helps mitigate the risk of a bad actor adding a backdoor to the action's repository, as they would need to generate a SHA-1 collision for a valid Git object payload. When selecting a SHA, you should verify it is from the action's repository and not a repository fork.

So, for now I suggest we keep the targeted commit.

File: .github/workflows/build-docker-image.yml (hunk 3 above)
Comment (apulsipher) [reply .github/workflows/build-docker-image.yml:42]: See above

File: .github/workflows/build-docker-image.yml (hunk 4 above)
Comment (apulsipher) [reply .github/workflows/build-docker-image.yml:50]: See above

File: .github/workflows/build-docker-image.yml (hunk 5 above)
Comment (apulsipher) [reply .github/workflows/build-docker-image.yml:63]: Yes. You can read about artifact attestations here: https://docs.github.com/en/actions/security-for-github-actions/using-artifact-attestations/using-artifact-attestations-to-establish-provenance-for-builds

**Review by gvegayon — APPROVED** [review]

**Pull request merged**